src/cljs/athens/page.cljs

```diff
@@ -70,14 +70,37 @@
        [:div {:style {:margin-left 20}}
         [render-blocks (:block/uid @node)]]])))
 
+(defn title [title]
+  (let [s (reagent/atom {:editing false
+                         :current-title title
+                         :new-title title})
+        save! (fn [] (dispatch [:node/rename (:current-title @s) (:new-title @s)])
+                     (swap! s assoc :editing false))
+        cancel! (fn [] (swap! s #(-> %
+                                     (assoc :editing false)
+                                     (assoc :new-title (:current-title @s)))))]
```

Comment:
I would leave the state management of the input to the browser. This gives a better user experience. You can access the current value via the event. This means you can remove the `:on-change` event
```clojure
save! (fn [event]
        (let [new-title (-> event .-target .-value)]
          (dispatch [:node/rename (:current-title @s) event-title])
          (swap! s assoc
                 :editing false
                 :current-title new-title)))
cancel! (fn [] (swap! s assoc :editing false))

Reply:
In what scenario would edit's be ignored you think? As a long as you have focus on that field (no escape, no blur), that value is correct, right?

I might be missing something as well though :) I might try it later and show what I mean if it doesn't make sense (maybe it doesn't)

Reply:
You were right, it was possuble to get rid of the `:on-change`.